lib/oci_utils/impl/__init__.py

```diff
@@ -213,25 +214,64 @@ def read_config():
     return oci_utils_config
 
 
-def setup_logging():
+class levelsFilter(logging.Filter):
+    """
+    By logging level filter.
+    the filter will discard any record that do not have the right level
+    """
+
+    def __init__(self, levels, name=''):
+        """
+        Constructs a new filter
+        levels : list of level (as int) that are accepted and will lead to
+          the actual handling of the record
+        """
+        logging.Filter.__init__(self, name)
+        self.levels = levels
+
+    def filter(self, record):
+        if record.levelno in self.levels:
+            return True
+        return False
+
+
+def setup_logging(forceDebug=False):
     """
-    General function to setup logger
+    General function to setup logger.
+    Everything from debug to stdout message is handle by loggers.
+    stdout logger handle info and warning message to STDOUT
+    stdder logger handle error and critical message to stderr
+    anything else is for debug logger which log everything to /var/tmp/oci-utils.log
     """
 
+    flatFormatter = logging.Formatter('%(message)s')
+    formatter = logging.Formatter(
+        '%(asctime)s - %(name)s - %(levelname)s(%(module)s:%(lineno)s) - %(message)s')
+
     if os.environ.get('_OCI_UTILS_SYSLOG'):
         handler = SysLogHandler(address='/dev/log',
                                 facility=SysLogHandler.LOG_DAEMON)
     else:
-        handler = logging.StreamHandler(stream=sys.stderr)
+        handler = logging.FileHandler('/var/tmp/oci-utils.log', mode='a',)
```

Comment:
Shouldn't this be RotatingFileHandler to prevent the file from just growing indefinitely?

Reply:
We can do that. but note that this file exists only if debug flag is on.